tao_compiler/mlir/disc/transforms/disc_lower_to_library_call.cc

```diff
@@ -463,6 +466,60 @@ Value getRootMemRefIfSafe(Value memref) {
   return rootMemRef;
 }
 
+struct TransposeConverter : public OpRewritePattern<lmhlo::TransposeOp> {
+  using OpRewritePattern<lmhlo::TransposeOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(lmhlo::TransposeOp op,
+                                PatternRewriter& rewriter) const override {
+    auto permutation = op.getPermutation().getValues<int64_t>();
+    int rank = permutation.size();
+    if (rank != 2) return failure();
+    // only rewriter custom library when switch 1 and 2 dimensions of
+    // a 3d tensor, that means permute = [0, 2, 1]
+    if (rank == 3 && permutation[1] != 2 && permutation[2] != 1)
+      return failure();
```

Comment:
It seems the second condition if will not be reached.

Reply:
done